TypeCobol/Compiler/FileCompiler.cs

```diff
@@ -154,41 +158,34 @@ public class FileCompiler
             chrono.Start();
             if (textDocument == null)
             {
-                TextDocument = new ReadOnlyTextDocument(sourceFile?.Name, sourceFile?.Encoding, columnsLayout, sourceFile?.ReadChars());
-            }
-            // 2.b Load it in an existing text document in memory
-            else if (sourceFile != null)
-            {
-                TextDocument = textDocument;
-                textDocument.LoadChars(sourceFile.ReadChars());
+                Debug.Assert(sourceFile != null);
+                Debug.Assert(fileInfo != null);
```

Comment:
`fileInfo` can be given with the value `null` to the function.
It would mean `sourceFile` will be null too.

Reply:
`fileInfo` can be `null` but in that case that would mean we have a non-null `textDocument`. Both `fileInfo` and `textDocument` describe the input to be parsed and one of them should be not null but not both. I will add a `Debug.Assert(textDocument == null)` in the first `if` to emphasize this. As for `sourceFile`, the assertion here is merely to avoid ReSharper warning because if we have a non-null `fileInfo` then it means we performed a search in `SourceFileProvider` and we found something otherwise we would have been in exception.

Reply:
See 48f2dc71c62515b29de1ede469594cd9623a410c.

Reply:
OK, this is easier to read this way.